lib/statsd.js

```diff
@@ -464,5 +462,37 @@ Client.prototype.childClient = function(options) {
   return new ChildClient(this, options);
 };
 
+// Overrides tags in parent with tags from child with the same name (case sensitive) and return the result as new
+// array. parent and child are not mutated.
+function overrideTags (parent, child) {
+  var childCopy = {};
+  var toAppend = [];
+  child.forEach(function (tag) {
+    var idx = tag.indexOf(':');
```

Comment:
There is always a way to "avoid" unexpected exceptions - check each and every parameter for correctness and try to do something meaningful with it. I think as a general approach to writing code it is a bad idea: 
# user of the library should use it correctly and if it is misused it should let the user know about it.
# the code becomes full of "safety checks" and it makes it hard to maintain.
# There is no guarantee all cases are handled anyway (bug in a library itself may cause an exception).

I agree that this changes the behaviour but I thought you wanted to release a new major version? :) This feature itself changes the behaviour for existing programs.

p.s. as a user of the library I'd prefer the library to fail fast and crash my program to let me know that I'm doing something stupid so that I can go and fix my crappy code, rather than masking the bug. Running program that produces no errors but incorrect results is a nightmare to debug, isn't it?